src/Renci.SshNet/BaseClient.cs

```diff
@@ -228,14 +228,23 @@ public void Connect()
             // forwarded port with a client instead of with a session
             //
             // To be discussed with Oleg (or whoever is interested)
-            if (IsSessionConnected())
+            if (IsConnected)
             {
                 throw new InvalidOperationException("The client is already connected.");
             }
 
             OnConnecting();
 
-            Session = CreateAndConnectSession();
+            // The session may already/still be connected here because e.g. in SftpClient, IsConnected also checks the internal SFTP session
+            var session = Session;
+            if (session is null)
+            {
+                Session = CreateAndConnectSession();
+            }
+            else if (!session.IsConnected)
+            {
+                session.Connect();
+            }
```

Comment:
Previously, if `Session.IsConnected` were false, we would create a new Session. I think we should keep that the same.

```suggestion
            var session = Session;
            if (session is null || !session.IsConnected)
            {
                Session = CreateAndConnectSession();
            }
```

Reply:
fixed